httomo/yaml_checker.py

```diff
@@ -216,114 +130,165 @@ def check_methods_exist_in_templates(conf: PipelineConfig) -> bool:
     Check if the methods in the pipeline YAML file are valid methods, by
     checking if they exist in the template YAML files.
     """
-    modules, methods, packages = _get_pipeline_info(conf)
-    template_yaml_files = _get_yaml_templates(modules, methods, packages)
+    packages = _get_package_info(conf)
+    template_yaml_files = _get_template_yaml(conf, packages)
 
     for i, f in enumerate(template_yaml_files):
         if not os.path.exists(f):
             _print_with_colour(
-                f"'{modules[i] + '/' + next(iter(methods[i]))}' is not a valid"
-                " path to a method. Please recheck the yaml file."
+                f"'{conf[i]['module_path'] + '/' + conf[i]['method']}' is not a valid"
+                " method. Please recheck the yaml file."
             )
             return False
 
     return True
 
 
-def check_valid_method_parameters(
-        conf: PipelineConfig,
-) -> bool:
+def check_parameter_names_are_known(conf: PipelineConfig) -> bool:
     """
-    Check each method config in the pipeline against the templates to see if
-    the given parameter names are valid.
+    Check if the parameter name of config methods exists in yaml template method parameters
     """
-    modules, methods, packages = _get_pipeline_info(conf)
-    template_yaml_files = _get_yaml_templates(modules, methods, packages)
-    template_yaml_data_list: List[Generator] = []
-
-    for f in template_yaml_files:
-        with open(f, "r") as template:
-            method_dict = yaml.load(template, Loader=yaml.FullLoader)[0]
-            template_yaml_data_list.append(
-                next(iter(method_dict.values()))
+    template_yaml_conf = _get_template_yaml_conf(conf)
+    for method_dict in conf:
+        yml_method_list = [
+            yml_method_dict
+            for yml_method_dict in template_yaml_conf
+            if (yml_method_dict["method"] == method_dict["method"])
+            & (yml_method_dict["module_path"] == method_dict["module_path"])
```

Comment:
A nitpicky point, but I think in python, the `&` operator is a [bitwise operator](https://docs.python.org/3/reference/expressions.html#binary-bitwise-operations) and is intended to be used on bit values.

Possibly due to some type coercion happening or something it still works as intended, but it may be worth changing the `&` to `and` to signal the intent that the code isn't expecting to be working on bit values here (at least, it doesn't look like it's expecting bit values, but please do correct me if I'm wrong)?

```suggestion
            and (yml_method_dict["module_path"] == method_dict["module_path"])
```

Reply:
Thank you for checking this and making the suggestion. Completed in [71641dd](https://github.com/DiamondLightSource/httomo/pull/201/commits/71641dd9f80f66081ba0c954586eaec17467ee42)